### ISSUE TITLE
Problem with acceptance test

### ISSUE THREAD

**Issue by littlejo** [opentofu/opentofu#1795]
### OpenTofu Version

* Last version of opentofu (1.6 and 1.7)
* but no problem before
* no problem with terraform.

### OpenTofu Configuration Files

NA

### Debug Output

NA

### Expected Behavior

https://github.com/littlejo/terraform-provider-cilium/actions/runs/9779727561

### Actual Behavior

https://github.com/littlejo/terraform-provider-cilium/actions/runs/9876343243
```
 install_test.go:14: Step 1/3 error: After applying this test step, the non-refresh plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # cilium.test will be updated in-place
          ~ resource "cilium" "test" {
                id          = "cilium"
                # (9 unchanged attributes hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

Cicd: https://github.com/littlejo/terraform-provider-cilium/blob/main/.github/workflows/test.yml#L72

**Comment by cam72cam** [comment]
So this is a new behavior as of the 1.7.3 and 1.6.3 release of OpenTofu?  It seems like the timing lines up.

Are you able to explicitly pin 1.7.2 and 1.6.2 in your config to confirm that it was the .2 to .3 upgrade?

**Comment by littlejo** [comment]
> Are you able to explicitly pin 1.7.2 and 1.6.2 in your pipeline to confirm that it was the .2 to .3 upgrade?

I just pinned 1.7.2 and 1.6.2 and I also have a problem. That's very strange because the CI doesn't have problem last week.

**Comment by cam72cam** [comment]
Hmm, would you be open to trying the .1 and .0 releases as well?  Perhaps the action was not pulling in .2 or .1 for some reason?

**Comment by littlejo** [comment]
> Hmm, would you be open to trying the .1 and .0 releases as well? 

Same problem on .1 and .0

**Comment by cam72cam** [comment]
That is incredibly strange.  You must be running the same OpenTofu code that you were using last week, but it now does not function.  Is it possible there are any implicit dependencies that have changed?

**Comment by littlejo** [comment]
I saw another thing, TestAccCiliumDeployResource passed. It is very similar as TestAccCiliumInstallResource which failed.
I created the deploy resource for Crossplane importation (crossplane upjet). Because install resource failed with importation.
The differences between Cilium Deploy and Cilium Install resource are:
* The name of resource: _cilium_deploy_ instead of _cilium_
* I removed one attribute in _cilium_deploy_ resource: ca is for retrieve certificates when installation of cilium is done.
```
var CaAttributeTypes = map[string]attr.Type{
	"crt": types.StringType,
	"key": types.StringType,
}

		Attributes: map[string]schema.Attribute{
			//"ca": schema.ObjectAttribute{
			//	AttributeTypes: CaAttributeTypes,
			//	Computed:       true,
			//	Sensitive:      true,
			//},
```

**Comment by littlejo** [comment]
Ok i found it!
In fact it's not a bug from tofu but from Terraform: Terraform is already installed on github action vm and was upgraded on the last version and the acceptance test use terraform instead of tofu.

So i have to change acceptance test with tofu (https://opentofu.org/docs/language/providers/#running-acceptance-tests-with-opentofu-cli) and i need to fix the problem for terraform 1.9 now.
